File: src/ReverseWords.java

public class ReverseWords {
	public static String reverseWords(String input) {
        if (input == null || input.length() == 0) {
			return "";
		}
		String[] words = input.split(" ");
		StringBuilder output = new StringBuilder();
		for (int index = words.length - 1; index >= 0; index --) {
			if (words[index].length() == 0)
				continue;
			output.append(words[index]);
			output.append(" ");
		}
		return output.toString().trim();
    }
	
	public static void main(String[] args) {
		System.out.println(reverseWords(" 1"));
	}
}

File: python/reverse_words.py
#! /usr/bin/pyhon

'''
Given an input string, reverse the string word by word.
For example,
Given s = "the sky is blue",
return "blue is sky the".
'''

class Solution:
    # @param s, a string
    # @return a string
    def reverseWords(self, s):
        return ' '.join(s.split()[::-1])

if __name__ == '__main__':
    solution = Solution()
    print solution.reverseWords('the sky is blue')
